relay-server/src/services/store.rs

```diff
@@ -812,6 +836,47 @@ impl StoreService {
         Ok(())
     }
 
+    #[allow(clippy::too_many_arguments)]
+    fn produce_replay_messages(
+        &self,
+        replay_event: Option<&Item>,
+        replay_recording: Option<&Item>,
+        replay_id: EventId,
+        scoping: Scoping,
+        start_time: Instant,
+        retention_days: u16,
+        send_combined_replay_envelope: bool,
+    ) -> Result<(), StoreError> {
+        if let Some(replay_event) = replay_event {
+            self.produce_replay_event(
+                replay_id,
+                scoping.organization_id,
+                scoping.project_id,
+                start_time,
+                retention_days,
+                replay_event,
+            )?;
+        }
+
+        if let Some(replay_recording) = replay_recording {
+            let combined_replay_event = if send_combined_replay_envelope && replay_event.is_some() {
+                replay_event
+            } else {
+                None
+            };
+
+            self.produce_replay_recording(
+                Some(replay_id),
+                scoping,
+                replay_recording,
+                combined_replay_event,
+                start_time,
+                retention_days,
+            )?;
+        }
```

Comment:
the `replay_event.is_some()` check seems unnecesary, if it's none you can just use the variable directly anyway 

```suggestion
    if let Some(replay_recording) = replay_recording {
        self.produce_replay_recording(
            replay_id,
            scoping,
            replay_recording,
            if send_combined_replay_envelope { replay_event } else { None },
            start_time,
            retention_days,
        )?;
    }
```